src/__tests__/utils/survey-event-receiver.test.ts

```diff
@@ -1,157 +1,323 @@
-                },
+                actions: [createAction(3, '$pageview') as unknown as ActionType],
             },
-        } as unknown as Survey,
-    ]
-
-    beforeEach(() => {
-        config = {
-            token: 'testtoken',
-            api_host: 'https://app.posthog.com',
-            persistence: 'memory',
-        } as unknown as PostHogConfig
-
-        instance = {
-            config: config,
-            persistence: new PostHogPersistence(config),
-        } as unknown as PostHog
-    })
+        } as unknown as Survey
 
-    afterEach(() => {
-        instance.persistence?.clear()
-    })
-
-    it('register makes receiver listen for all surveys with events', () => {
-        const surveyEventReceiver = new SurveyEventReceiver(instance.persistence)
-        surveyEventReceiver.register(surveysWithEvents)
-        const registry = surveyEventReceiver.getEventRegistry()
-        expect(registry.has('second-survey')).toBeFalsy()
-        expect(registry.has('first-survey')).toBeTruthy()
-        expect(registry.get('first-survey')).toEqual([
-            'user_subscribed',
-            'user_unsubscribed',
-            'billing_changed',
-            'billing_removed',
-        ])
-
-        expect(registry.has('third-survey')).toBeTruthy()
-        expect(registry.get('third-survey')).toEqual(['user_subscribed', 'user_unsubscribed', 'address_changed'])
-    })
-
-    it('receiver activates survey on event', () => {
-        const surveyEventReceiver = new SurveyEventReceiver(instance.persistence)
-        surveyEventReceiver.register(surveysWithEvents)
-        surveyEventReceiver.on('billing_changed')
-        const activatedSurveys = surveyEventReceiver.getSurveys()
-        expect(activatedSurveys).toContain('first-survey')
-    })
+        it('can match action on event name', () => {
+            const myPageViewSurvey = {
+                name: 'my pageview survey',
+                id: 'my-pageview-survey',
+                description: 'pageview survey description',
+                type: SurveyType.Popover,
+                questions: [{ type: SurveyQuestionType.Open, question: 'what is a bokoblin?' }],
+                conditions: {
+                    actions: {
+                        values: [createAction(3, '$mypageview') as unknown as ActionType],
+                    },
+                },
+            } as unknown as Survey
+            autoCaptureSurvey.conditions.actions.values = [createAction(2, '$match_event_name')]
+            const surveyEventReceiver = new SurveyEventReceiver(instance)
+            console.log(`TEST:: registering surveys to observe`)
+            surveyEventReceiver.register([autoCaptureSurvey, myPageViewSurvey])
+            surveyEventReceiver._getActionMatcher().on('$match_event_name', createCaptureResult('$match_event_name'))
+            console.log(`TEST:: getting surveys to test`)
+            expect(surveyEventReceiver.getSurveys()).toEqual(['first-survey'])
```

Comment:
For better readability, I'd prefer the id of `autoCaptureSurvey` to match the variable name, i.e., `auto-capture-survey`.